src/macaron/slsa_analyzer/git_url.py

```diff
@@ -157,7 +154,10 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "")
             return False
 
     if digest:
-        if head_commit.hexsha == digest:
+        current_head: Commit = git_obj.repo.head.commit
```

Comment:
> If current_head is None here, should we return False early or should we proceed to try to check out the specific commit by the user thinking . The Exceptions for checking if digest exists or whether we can checkout digest (after [here](https://github.com/oracle-samples/macaron/blob/117-macaron-pull-the-latest-when-analyzing-a-local-cloned-repository/src/macaron/slsa_analyzer/git_url.py#L159)) should be caught and return False.

What I meant is that in this line, there might be the case where `current_head` is None (there might be some errors with extracting the head commit). Should we exit on error early here, or we ignore it to at least try to check out the specific digest down below (where any exception would be caught).

Reply:
It should be safe to try again to check out the `digest` if it exists, because `commit_exists` should return gracefully if commit doesn't exist.